### PR TITLE
General documentation cleanup for June - Part 1

### PR DESCRIPTION
**Pull request by lkasso**
### Details
This is a Documentation change ONLY.

### Context
This is an update for the User Manual: **---> [BUILD LINK FOR REVIEW](https://builds.gradle.org/repository/download/Gradle_Release_Check_BuildDistributions/83549138:id/distributions/gradle-8.9-docs.zip!/gradle-8.9-20240612211232%2B0000/docs/userguide/userguide.html) <---**
1.  Fix https://github.com/gradle/gradle/issues/29394
2. Fix https://github.com/gradle/gradle/issues/28851
3. Fix https://github.com/gradle/gradle/issues/27694
4. Fix https://github.com/gradle/gradle/issues/28231 (more details)
5. Move "Gradle on CI" to the community site (remove from this repo)
6. Remove the Gradle Deamon profile tip `--profile` for @runningcode
7. Add the Gradle Cookbook URL to the side menu of the User Manual and to the "Third Party Integration" page
8. Update deleted pages to redirect + update any references accordingly (for example [authoring_maintainable_build_scripts.adoc](https://github.com/gradle/gradle/blob/lkasso/documentation/june-cleanup/platforms/documentation/docs/src/docs/userguide/redirects/authoring_maintainable_build_scripts.adoc) and [PublicApi.kt](https://github.com/gradle/gradle/blob/e6ae406e742b748929e1b5c34b68f861a3028384/build-logic-commons/basics/src/main/kotlin/gradlebuild/basics/PublicApi.kt))

### CODE REVIEW

**Comment by lkasso** [comment]
@bot-gradle test BD

File: platforms/documentation/docs/src/docs/userguide/redirects/github-actions.adoc [hunk 1]
```diff
@@ -0,0 +1,3 @@
+++++
+<meta http-equiv="refresh" content="0;URL=https://community.gradle.org/">
```
Comment (oleg-nenashev) [inline platforms/documentation/docs/src/docs/userguide/redirects/github-actions.adoc:2]: IIUC it is not a redirect to the page but to the root, right?

File: platforms/documentation/docs/src/docs/userguide/reference/third_party_integration.adoc [hunk 2]
```diff
@@ -15,139 +15,4 @@
 [[third_party_integration]]
 = Gradle & Third-party Tools
 
-Gradle can be integrated with many different third-party tools such as IDEs and continuous integration platforms. Here we look at some of the more common ones as well as how to integrate your own tool with Gradle.
-
-[[ides]]
-== IDEs
```
Comment (oleg-nenashev) [inline platforms/documentation/docs/src/docs/userguide/reference/third_party_integration.adoc:21]: Are you moving it to the Cookbook? AFAICT we do not have a replacement content
As of now, I suggest incorporating the difference to https://docs.gradle.org/current/userguide/gradle_ides.html or moving the pages to the Cookbook and putting references

File: platforms/documentation/docs/src/docs/userguide/reference/third_party_integration.adoc [hunk 3]
```diff
@@ -15,139 +15,4 @@
...
-
-There are two main ways to integrate a tool with Gradle:
-
- * The Gradle build uses the tool
- * The tool executes the Gradle build
-
-The former case is typically <<custom_plugins.adoc#custom_plugins,implemented as a Gradle plugin>>. The latter can be accomplished by embedding Gradle through the Tooling API as described below.
-
-[[embedding]]
-== Embedding Gradle using the Tooling API
```
Comment (oleg-nenashev) [inline platforms/documentation/docs/src/docs/userguide/reference/third_party_integration.adoc:71]: Is there a replacement page for Tooling API information somewhere? AFAICT it is still fully supported https://docs.gradle.org/current/javadoc/org/gradle/tooling/GradleConnector.html

File: platforms/documentation/docs/src/docs/userguide/reference/third_party_integration.adoc [hunk 4]
```diff
@@ -15,139 +15,4 @@
...
-====
-
-The main entry point to the Tooling API is the link:{javadocPath}/org/gradle/tooling/GradleConnector.html[GradleConnector].
-You can navigate from there to find code samples and explore the available Tooling API models.
-You can use link:{javadocPath}/org/gradle/tooling/GradleConnector.html#connect--[GradleConnector.connect()] to create a link:{javadocPath}/org/gradle/tooling/ProjectConnection.html[ProjectConnection].
-A `ProjectConnection` connects to a single Gradle project.
-Using the connection you can execute tasks, tests and retrieve models relative to this project.
-
-[[sec:embedding_compatibility]]
-=== Compatibility of Java and Gradle versions
```
Comment (oleg-nenashev) [inline platforms/documentation/docs/src/docs/userguide/reference/third_party_integration.adoc:131]: Should the content be merged into https://docs.gradle.org/current/userguide/compatibility.html#java

File: platforms/documentation/docs/src/docs/userguide/reference/third_party_integration.adoc [hunk 5]
```diff
@@ -15,139 +15,4 @@
...
-The rule for the lower bound is the following:
-
-- Gradle 3.x and 4.x require a minimum version of Java 7.
-- Gradle 5 and above require a minimum version of Java 8.
-
-The Tooling API version is guaranteed to support running builds with all Gradle versions for the last five major releases.
-For example, the Tooling API 8.0 release is compatible with Gradle versions >= 3.0.
-Besides, the Tooling API is guaranteed to be compatible with future Gradle releases for the current and the next major.
-This means, for example, that the 8.1 version of the Tooling API will be able to run Gradle 9.x builds and _might_ break with Gradle 10.0.
+This page has been removed.
```
Comment (oleg-nenashev) [inline platforms/documentation/docs/src/docs/userguide/reference/third_party_integration.adoc:18]: Should it actually reference some page?

File: platforms/documentation/docs/src/main/resources/header.html [hunk 6]
```diff
@@ -364,15 +363,6 @@ <h3 id="authoring-gradle-builds-native">Authoring C++/Swift Builds</h3>
                 <li><a href="../userguide/swift_testing.html">Testing Swift projects</a></li>
             </ul>
 
-            <h3 id="gradle-on-ci">Gradle on CI</h3>
```
Comment (oleg-nenashev) [inline platforms/documentation/docs/src/main/resources/header.html:367]: Should we create a link to Gradle Cookbook in the User Guide navbar?

**Review by oleg-nenashev — CHANGES_REQUESTED** [review]
One of the biggest changes in this PR is removal of https://docs.gradle.org/current/userguide/third_party_integration.html , with no links to the replacement content, and with no merge into existing pages. This removal is not listed in the scope of the PR, so I am not sure what is the reasoning for removing content from this page without equal replacement.

I would suggest the following:

- Move the IDE integrations content as a new section in the cookbook
- Extract Tooling API somewhere, AFAICT it is still a part of the user guide (or maybe a developer guide)
- Nice2Have: Fix redirects to the links so they point to the pages

**Comment by lkasso** [comment]
@bot-gradle test BD

**Comment by lkasso** [comment]
@bot-gradle test BD

**Comment by lkasso** [comment]
@bot-gradle clean

**Comment by lkasso** [comment]
@bot-gradle test BD

**Comment by lkasso** [comment]
@bot-gradle clean

**Review by oleg-nenashev — APPROVED** [review]
I would suggest starting the Cookbook section as in https://github.com/gradle/gradle/pull/29430#discussion_r1636059376 , but it can be done later once we have more content there

**Comment by lkasso** [comment]
> I would suggest starting the Cookbook section as in [#29430 (comment)](https://github.com/gradle/gradle/pull/29430#discussion_r1636059376) , but it can be done later once we have more content there

I'm not ready for that yet. I prefer to do it in Version 2.0 of the manual.

**Pull request merged**